compiler/src/typed/typecore.re

```diff
@@ -2866,27 +2997,40 @@ let report_error = (env, ppf) =>
         );
       };
     }
-  | Too_many_arguments(in_function, ty, explanation) => {
+  | Not_a_function(ty, explanation) => {
       reset_and_mark_loops(ty);
-      if (in_function) {
-        fprintf(ppf, "This function expects too many arguments,@ ");
+      fprintf(ppf, "This expression is not a function,@ ");
+      fprintf(
+        ppf,
+        "the expected type is@ %a%t",
+        type_expr,
+        ty,
+        report_type_expected_explanation_opt(explanation),
+      );
+    }
+  | Function_label_mismatch({got, expected, expected_type, explanation}) => {
+      reset_and_mark_loops(expected_type);
+      if (is_optional(got)) {
         fprintf(
           ppf,
-          "it should have type@ %a%t",
-          type_expr,
-          ty,
-          report_type_expected_explanation_opt(explanation),
+          "This function contains the optional, default argument %s@ ",
+          qualified_label_name(got),
         );
       } else {
-        fprintf(ppf, "This expression should not be a function,@ ");
         fprintf(
           ppf,
-          "the expected type is@ %a%t",
-          type_expr,
-          ty,
-          report_type_expected_explanation_opt(explanation),
+          "The expected function type contains the optional, default argument %s@ ",
+          qualified_label_name(expected),
         );
       };
+      fprintf(ppf, "but the matching argument is not optional.@ ");
```

Comment:
```suggestion
      fprintf(ppf, "but the matching argument is not defaulted.@ ");
```

Reply:
I think `but the matching argument does not have a default value` would be much better then using the word `defaulted` I dont think `optional` worked terribly though but that would require us making it clear that the only way to get an optional argument is by having a default value.